src/core/client.py

```diff
@@ -118,10 +120,23 @@ def get_username(self):
             username = response["username"]
             return username
         except KeyError:
-            raise ValueError(
-                "The token did not return a valid username.",
-                response,
-            )
+            logger.warn(f"Error response: {response}")
+            raise InvalidToken()
+
+    def get_is_autoconfirmed(self):
+        response = self.get(self.ENDPOINT_PROFILE).json()
+        try:
+            groups = response["groups"]
+            return "autoconfirmed" in groups
+        except KeyError:
+            logger.warn(f"Error response: {response}")
+            raise InvalidToken()
+
+    def is_autoconfirmed_or_raise(self):
```

Comment:
This method is strange....if the user is autoconfirmed, it should return True. And by opposition, if the user is not autoconfirmed, it should return False. 

It is only used once, so maybe its code should go directly to the block that verifies it the user is autoconfirmed and raises an exception otherwise.